coral/src/app/layout/main-navigation/MainNavigation.test.tsx

```diff
@@ -0,0 +1,226 @@
+      const nav = screen.getByRole("navigation", { name: "Main navigation" });
+      expect(nav).toBeVisible();
+    });
+
+    navLinks.forEach((link) => {
+      it(`renders a link for ${link.name}`, () => {
+        const nav = screen.getByRole("navigation", {
+          name: "Main navigation",
+        });
+
+        const navLink = within(nav).getByRole("link", { name: link.name });
+        expect(navLink).toBeVisible();
+        expect(navLink).toHaveAttribute("href", link.linkTo);
+      });
+    });
+
+    it(`renders all navigation items`, () => {
+      const navLinks = screen.getAllByRole("link");
+
+      expect(navLinks).toHaveLength(navLinks.length);
+    });
+
+    submenuItems.forEach((submenu) => {
+      it(`renders a button to open submenu for ${submenu.name}`, () => {
+        const nav = screen.getByRole("navigation", {
+          name: "Main navigation",
+        });
+
+        const button = within(nav).getByRole("button", {
+          name: `${submenu.name} submenu, closed. Click to open.`,
+        });
+
+        expect(button).toBeEnabled();
+      });
+    });
+
+    it(`renders all submenu buttons`, () => {
+      const submenuItems = screen.getAllByRole("button");
+
+      expect(submenuItems).toHaveLength(submenuItems.length);
+    });
+
+    // temp for special handling of "Topics"
+    submenuItemTopics.forEach((submenu) => {
+      it(`renders a button to close submenu for ${submenu.name}`, () => {
+        const nav = screen.getByRole("navigation", {
+          name: "Main navigation",
+        });
+
+        const button = within(nav).getByRole("button", {
+          name: `${submenu.name} submenu, open. Click to close.`,
+        });
+
+        expect(button).toBeEnabled();
+      });
+
+      it(`renders a submenu list for ${submenu.name}`, () => {
+        const list = screen.getByRole("list", {
+          name: `${submenu.name} submenu`,
+        });
+        expect(list).toBeVisible();
+      });
+
+      submenu.links.forEach((link) => {
+        it(`renders the submenu links ${link}`, () => {
+          const menu = screen.getByRole("list", {
+            name: `${submenu.name} submenu`,
+          });
+          const submenuLink = within(menu).getByRole("link", { name: link });
+          expect(submenuLink).toBeVisible();
+        });
+      });
+    });
+
+    it(`renders icons for all nav links that are hidden from assistive technology`, () => {
+      // every navlink and submenu link has one icon
+      // every submenu link has an icon to indicate opened/closed
+      const iconAmount =
+        navLinks.length +
+        submenuItems.length * 2 +
+        submenuItemTopics.length * 2;
+      const nav = screen.getByRole("navigation", {
+        name: "Main navigation",
+      });
+
+      const icons = within(nav).getAllByTestId("ds-icon");
+      expect(icons).toHaveLength(iconAmount);
+    });
+  });
+
+  describe("user can open submenus and see more links", () => {
```

Comment:
Should we also have test cases for keyboard navigation, so we don't only test click events?

Reply:
🤯 🤯 🤯 🤯  YES of course, that's a great improvement! ❤️

Reply:
I'll also add such a case in my PR ^^

Reply:
updated! 

I could not cover the skipping to main content in `Layout` 😞  because I couldn't get the mocks for ref right 🤔